docs: review bt mesh models (#2504)
nrfconnect/sdk-nrf: include/bluetooth/mesh/gen_prop_srv.rst

Generic Property Servers
########################

A Generic Property Server holds a list of device properties.

> The following different property servers are available:

Reviewer: This change was a bit awkward I think. I get the motivation behind removing the count, but in this case, it's very unlikely that it'll ever change. I think the old text was better, but perhaps removing "different" would be an okay middle ground?

- Generic Manufacturer Property Server - Provides access to Manufacturer properties, which are read-only properties set by the device manufacturer.
- Generic Admin Property Server - Provides access to Admin properties, which can be read and written to.
- Generic User Property Server - Provides access to both Admin and Manufacturer properties on its element, but only if the Property owner allows it.
- Generic Client Property Server - Provides a list of all properties supported by the Generic Property Client on the same element.
  The Client Property Server does not actually own any properties, and does not need the user to provide access to any property values.

> When developing, take into account the following information about the servers:

Reviewer: ```suggestion
When developing, you should take the following into account:
```

* The set of Property IDs and their order cannot be changed.

States
======

The Generic Power OnOff Server model can only assume the one state described in this section.

Device Properties: :cpp:type:`bt_mesh_prop`
    A single server may own several Device Properties, which are accessed by the Property ID.
    Each Property holds a value and a user access parameter, which controls the Property's availability to a Generic User Property Server on the same element.

    The Device properties have slightly different access restrictions depending on the type of server that owns them:

    * Manufacturer Property - The value can only be read.
    * Admin Property - The value can be read and written.
    * Client Property - Has no associated value, and is only a listing of the property IDs supported by the accompanying Generic Property Client.

Extended models
===============

The Generic Admin Property Server and Generic Manufacturer Property Server should both extend the Generic User Property Server.
This is not handled automatically by the model implementation.

> For this reason, to ensures compliance with the Bluetooth Mesh Model specification, a Generic User Property Server must be instantiated in the same element in the device composition data as the Generic Admin or Generic Manufacturer Property server.

Reviewer: ```suggestion
For this reason, to ensure compliance with the Bluetooth Mesh Model specification, a Generic User Property Server must be instantiated in the same element in the device composition data as the Generic Admin or Generic Manufacturer Property server.
```
